gef.py

```diff
@@ -1003,19 +1003,25 @@ def psprint(self):
         return "\n".join(msg) + "\n"
 
 
+pattern_libc_ver = re.compile(b"glibc (\d+)\.(\d+)")
+
 @lru_cache()
 def get_libc_version():
     sections = get_process_maps()
-    try:
-        for section in sections:
-            if "libc" in section.path:
+    libc_version = 0, 0
+    for section in sections:
+        if "libc" in section.path:
+            try:
                 libc_version = tuple(int(_) for _ in
                                      re.search(r"libc6?[-_](\d+)\.(\d+)\.so", section.path).groups())
-                break
-        else:
-            libc_version = 0, 0
-    except AttributeError:
-        libc_version = 0, 0
+            except AttributeError:
+                if os.access(section.path, os.R_OK):
```

Comment:
why do you need this? You can just catch the exception if the file doesn't open

Reply:
To avoid exception handling, there are so many of them in this piece of code and it seemed ugly to me. Fix it anyway?